source/common/router/router.cc

```diff
@@ -1003,6 +1009,11 @@ void Filter::onUpstreamHeaders(uint64_t response_code, Http::HeaderMapPtr&& head
 
   modify_headers_(*headers);
 
+  // TODO: Handle the gRPC-status sent via trailers for outlier detection.
```

Comment:
Remove TODO from here. This fix is definitely required. 

===

The trailer status needs be addressed separately which may require adding something like OutlierDetector::putTrailerStatusCode() because we shouldn't 1) delay recording the initial ok status; and 2) overwriting the initial ok status with an error status.

@htuch  should we file a separate issue for the trailer status handling with OutlierDetector?

Reply:
Done.